implementations/rust/ockam/ockam_identity/src/secure_channel/handshake/handshake_worker.rs

```diff
@@ -287,7 +290,14 @@ impl HandshakeWorker {
     async fn handle_decrypt(&mut self, context: &mut Context, message: Routed<Any>) -> Result<()> {
         let decryptor_handler = self.decryptor_handler.as_mut().unwrap();
         let msg_addr = message.msg_addr();
-        if msg_addr == self.addresses.decryptor_remote {
+
+        if self.key_exchange_only {
```

Comment:
Is that intended that API won't work for regular secure channels? Also, I would consider not creating that mailbox in the first place

Reply:
My bad, but maybe it could be a little bit more easier to read if that check would be inside the existing `} else if msg_addr == self.addresses.decryptor_remote {`